src/DataCollection.Shared.Tests/IdentifyResultViewModelTests.cs

```diff
@@ -0,0 +1,426 @@
+            Assert.Contains(nameof(identifyVM.CurrentlySelectedFeature), changedProperties);
+
+            // Clear existing results and set new list with multiple items
+            changedProperties.Clear();
+            _sharedResultList = new List<IdentifiedFeatureViewModel>();
+            AddDummyResults(8);
+
+            identifyVM.SetNewIdentifyResult(_sharedResultList);
+
+            Assert.Contains(nameof(identifyVM.IdentifiedFeatures), changedProperties);
+            Assert.Contains(nameof(identifyVM.ResultCount), changedProperties);
+            Assert.False(changedProperties.Contains(nameof(identifyVM.CurrentFeatureIndex)));
+            Assert.False(changedProperties.Contains(nameof(identifyVM.CurrentlySelectedFeature)));
+        }
+
+        /// <summary>
+        /// Verifies that setting the selected feature index to null clears feature selection.
+        /// </summary>
+        /// <param name="resultCount">Number of identified results to test with.</param>
+        [TestCase(0)]
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        public void SettingNullFeatureIndexClearsSelection(int resultCount)
+        {
+            // Set up test
+            AddDummyResults(resultCount);
+            IdentifyResultViewModel identifyVM = new IdentifyResultViewModel();
+            identifyVM.SetNewIdentifyResult(_sharedResultList);
+
+            if (resultCount > 0)
+            {
+                identifyVM.CurrentFeatureIndex = resultCount - 1;
+                Assert.AreEqual(identifyVM.CurrentlySelectedFeature, _sharedResultList.ElementAt(resultCount - 1));
+            }
+
+            identifyVM.CurrentFeatureIndex = null;
+            Assert.IsNull(identifyVM.CurrentlySelectedFeature);
+        }
+
+        /// <summary>
+        /// Verifies that exceptions are thrown when invalid feature index is set.
+        /// </summary>
+        /// <param name="resultCount">Number of identified results to test with.</param>
+        [TestCase(0)]
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        public void SetInvalidFeatureIndex(int resultCount)
+        {
+            // Set up test
+            AddDummyResults(resultCount);
+            IdentifyResultViewModel identifyVM = new IdentifyResultViewModel();
+            identifyVM.SetNewIdentifyResult(_sharedResultList);
+
+            // -1 can't throw because that breaks UWP
+            Assert.DoesNotThrow(() =>
+            {
+                identifyVM.CurrentFeatureIndex = -1;
+                Assert.AreEqual(identifyVM.CurrentFeatureIndex, null);
+            });
+
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                identifyVM.CurrentFeatureIndex = resultCount;
+            });
+
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                identifyVM.CurrentFeatureIndex = resultCount + 1;
+            });
```

Comment:
Is this needed?

Reply:
I think these two asserts would be useful for preventing errors that might arise from e.g. accidentally changing from a `<=` comparison to a `<` or `=` comparison.